### PR TITLE
Use JSON-LD keywords for languages.

### PR DESCRIPTION
**Pull request by BigBlueHat**
This PR uses the non-aliased values for `@language`, `@direction`, and `@value` to prevent confusion that these aliased terms may be used anywhere for any purpose--instead highlighting their sole application for providing language metadata for a single string value.

***
<a href="https://pr-preview.s3.amazonaws.com/BigBlueHat/vc-data-model/pull/1340.html" title="Last updated on Nov 20, 2023, 2:37 PM UTC (6c008dc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1340/3fb62f8...BigBlueHat:6c008dc.html" title="Last updated on Nov 20, 2023, 2:37 PM UTC (6c008dc)">Diff</a>

### CODE REVIEW

**Review by dlongley — APPROVED** [review]

**Review by OR13 — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -5610,21 +5613,31 @@ <h3>Language and Base Direction</h3>
...
+  "@direction": "<code>DIRECTION</code>"
 }
         </pre>
 
+        <p>
+When the language value object is used in place of a string value, the object
+MUST contain a `@value` property whose value is a string, and SHOULD contain a
+`@language` property whose value is a string containing a well-formed
+`Language-Tag` as defined by [[BCP47]], and MAY contain a `@direction` property
+whose value is a text direction string defined by the `@direction` attribute in
```
Comment (aphillips) [inline index.html:5627]: Consider linking "text direction" to the definition in the I18N glossary.

File: index.html [hunk 2]
```diff
@@ -5610,21 +5613,31 @@ <h3>Language and Base Direction</h3>
...
+  "@value": "<span class="highlight">The string value</span>",
+  "@language": "<code>LANGUAGE</code>"
+  "@direction": "<code>DIRECTION</code>"
 }
         </pre>
 
+        <p>
+When the language value object is used in place of a string value, the object
+MUST contain a `@value` property whose value is a string, and SHOULD contain a
+`@language` property whose value is a string containing a well-formed
```
Comment (aphillips) [inline index.html:5625]: Consider making "well-formed" be "valid" here. Conformance should only require well-formed, but authors should choose valid language tags. If it is too difficult to make the emit/consume distinction here, this text is fine.

File: index.html [hunk 3]
```diff
@@ -5635,9 +5648,9 @@ <h3>Language and Base Direction</h3>
 
         <pre class="example nohighlight" title="Arabic text with a base direction of right-to-left">
 "title": {
-  "value": "<span class="highlight">HTML و CSS: تصميم و إنشاء مواقع الويب</span>",
-  "lang": "<code>ar</code>"
-  "dir": "<code>rtl</code>"
+  "@value": "<span class="highlight">HTML و CSS: تصميم و إنشاء مواقع الويب</span>",
```
Comment (aphillips) [inline index.html:5651]: You want a `dir` on the span here so the string displays RTL correctly :-)

```suggestion
  "@value": "<span class="highlight" dir="rtl" lang="ar">HTML و CSS: تصميم و إنشاء مواقع الويب</span>",
```

File: index.html [hunk 4]
```diff
@@ -5648,32 +5661,33 @@ <h3>Language and Base Direction</h3>
...
-  "@id": "https://www.w3.org/2018/credentials/examples#title"
-}
+        <pre class="example nohighlight" title="Multiple language texts provided for title">
+"title": [
+  {
+    "@value": "<span class="highlight">HTML and CSS: Designing and Creating Websites</span>",
+    "@language": "<code>en</code>"
+  },
+  {
+    "@value": "<span class="highlight">HTML و CSS: تصميم و إنشاء مواقع الويب</span>",
```
Comment (aphillips) [inline index.html:5675]: see above

File: index.html [hunk 5]
```diff
@@ -5648,32 +5661,33 @@ <h3>Language and Base Direction</h3>
...
 
         <p>
-When multiple languages, base directions, and annotations are used in a single
-natural language string, more complex mechanisms are typically required. It is
-possible to use markup languages, such as HTML, to encode text with multiple
-languages and base directions. It is also possible to use the
+When a single natural language string contains multiple languages, base
+directions, and annotations, a more complex mechanism is typically required.
+It is possible to use markup languages, such as HTML, to encode text with
+multiple languages and base directions. It is also possible to use the
```
Comment (aphillips) [inline index.html:5690]: The text here about base direction is wrong. Any paragraph--even one in multiple languages and scripts--has only one base paragraph direction. Inline directional changes that are part of the paragraph do exist and can require care on the part of the author, but this is outside the scope of string metadata. Perhaps:

```suggestion
When a single natural language string contains multiple languages or annotations,
the contents of the string might require additional structure or markup in order to be
presented correctly. It is possible to use markup languages, such as HTML, to label spans of text in different languages or to supply string-internal markup needed for proper display of <a>bidirectional text</a>. It is also possible to use the
```

**Review by aphillips — CHANGES_REQUESTED** [review]
Good start.

Where is the information on credential-level default language/direction (for cases where `name` and `direction` are just strings)?

File: index.html (hunk 2 above)
Comment (msporny) [inline index.html:5625]: The change below applies @aphillips request:

```suggestion
`@language` property whose value is a string containing a valid
```

We are trying very hard to stay away from any usage of "well-formed" text in the spec.

File: index.html (hunk 3 above)
Comment (msporny) [reply index.html:5651]: > You want a `dir` on the span here so the string displays RTL correctly :-)

Nope, that "highlight" is ReSpec markup on the example itself, not the markup that will be displayed in the spec. :)

We're highlighting the text that `@language` and `@direction` is being applied to... we're not using HTML to express `lang` and `dir`... and yes, it's difficult to see that just by staring at the source code in the spec. :)

File: index.html (hunk 4 above)
Comment (msporny) [reply index.html:5675]: See response to "see above" :) -- https://github.com/w3c/vc-data-model/pull/1340/files#r1390255038

File: index.html (hunk 3 above)
Comment (aphillips) [reply index.html:5651]: Not sure I understand. The `<span>` does end up in the rendered page (Respec is using it to style the string). By adding the dir and lang it makes the string display correctly (as if the @language and @direction had been applied):

image

File: index.html (hunk 2 above)
Comment (aphillips) [reply index.html:5625]: "well-formed" and "valid" are defined terms in BCP47 (see [here](https://www.rfc-editor.org/rfc/rfc5646.html#section-2.2.9)). They are also defined terms in the i18n glossary. Generally VC shouldn't require implementations to check for validity, because that involves reading the IANA Language Subtag Registry--you actually want conformance to hinge on well-formed language tags. But you wan users to pick language tags that are valid.

In this case, though, I can agree with the change provided you're aware of what you want to require people to demonstrate.

File: index.html (hunk 1 above)
Comment (BigBlueHat) [reply index.html:5627]: @aphillips there's no discrete "text direction" definition in the I18N glossary. Would you recommend I reference [base direction](https://www.w3.org/TR/i18n-glossary/#dfn-base-direction) or [paragraph direction](https://www.w3.org/TR/i18n-glossary/#dfn-paragraph-direction) instead?

Comment (aphillips) [reply index.html:5627]: @BigBlueHat Use base direction here.

File: index.html (hunk 4 above)
Comment (BigBlueHat) [reply index.html]: Fixed as @aphillips suggested.

File: index.html (hunk 2 above)
Comment (BigBlueHat) [reply index.html]: k. Putting this back to "well-formed".

File: index.html [hunk 6]
```diff
@@ -5635,45 +5648,45 @@ <h3>Language and Base Direction</h3>
 
         <pre class="example nohighlight" title="Arabic text with a base direction of right-to-left">
 "title": {
-  "value": "<span class="highlight">HTML و CSS: تصميم و إنشاء مواقع الويب</span>",
-  "lang": "<code>ar</code>"
-  "dir": "<code>rtl</code>"
+  "@value": "<span class="highlight">HTML و CSS: تصميم و إنشاء مواقع الويب</span>",
```
Comment (aphillips) [inline index.html:5651]: You didn't implement the fix here? (It is correct on line 5675)

**Comment by BigBlueHat** [comment]
> Where is the information on credential-level default language/direction (for cases where `name` and `direction` are just strings)?

It's clear you found #1339. 😃 

Thanks for all the help on these, @aphillips and @msporny. All your requests should be addressed now. Cheers!

File: index.html (hunk 6 above)
Comment (BigBlueHat) [reply index.html:5651]: I did in https://github.com/w3c/vc-data-model/pull/1340/commits/0222763398dd0d2d505ad570598e0a10b05aa410

Comment (BigBlueHat) [reply index.html:5651]: Oh. Missed the other line. My bad. Fixing.

**Review by aphillips — APPROVED** [review]
Looks good! Thanks @BigBlueHat

**Review by dlongley — APPROVED** [review]

**Review by brianorwhatever — APPROVED** [review]

**Comment by msporny** [comment]
This needs a rebase before I can merge it, @BigBlueHat.

**Comment by BigBlueHat** [comment]
@msporny rebased and ready when you are.

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.
